### PR TITLE
rules: Handle license categories 'proprietary-free' and 'free-restricted'

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

Context: #66.

### CODE REVIEW

File: evaluator-rules/src/main/resources/evaluator.rules.kts [hunk 1]
```diff
@@ -1002,8 +1002,9 @@ fun PackageRule.packageManagerSupportsDeclaredLicenses(): RuleMatcher =
  * Policy rules
  */
 
-fun RuleSet.copyleftInDependencyRule() = dependencyRule("COPYLEFT_IN_DEPENDENCY") {
+fun RuleSet.copyleftInDependencyRule() = packageRule("COPYLEFT_IN_DEPENDENCY") {
```
Comment (sschuberth) [inline evaluator-rules/src/main/resources/evaluator.rules.kts:1022]: IMO, even if a breaking change, the rule name and extension function name should then be adjusted accordingly.

Comment (sschuberth) [reply evaluator-rules/src/main/resources/evaluator.rules.kts:1022]: Just to given an example, e.g. here the naming is consistent:

    fun RuleSet.vulnerabilityInPackageRule() = packageRule("VULNERABILITY_IN_PACKAGE")

File: evaluator-rules/src/main/resources/evaluator.rules.kts [hunk 2]
```diff
@@ -1242,6 +1251,27 @@ fun RuleSet.packageCurationInOrtYmlRule() = ortResultRule("PACKAGE_CURATION_IN_O
     }
 }
 
+fun RuleSet.proprietaryFreeInDependencyRule() = packageRule("PROPRIETARY_FREE_IN_DEPENDENCY") {
```
Comment (sschuberth) [inline evaluator-rules/src/main/resources/evaluator.rules.kts:1282]: Also here, neither the function nor the rule name should use the word "dependency".

File: evaluator-rules/src/main/resources/evaluator.rules.kts [hunk 3]
```diff
@@ -1144,6 +1152,27 @@ fun RuleSet.deprecatedScopeExludeInOrtYmlRule() = ortResultRule("DEPRECATED_SCOP
     }
 }
 
+fun RuleSet.freeRestrictedInDependencyRule() = packageRule("FREE_RESTRICTED_IN_DEPENDENCY") {
```
Comment (sschuberth) [inline evaluator-rules/src/main/resources/evaluator.rules.kts:1155]: Also here, neither the function nor the rule name should use the word "dependency".

File: evaluator-rules/src/main/resources/evaluator.rules.kts (hunk 1 above)
Comment (fviernau) [reply evaluator-rules/src/main/resources/evaluator.rules.kts:1022]: To me having the names for these two consistent seems more important, which is why I prefer to
have the name end with `_IN_DEPENDENCY` for both. IMO whether the rule is a package- or dependency rule is an implementation detail which needs not to be reflected in the rule name.

```
fun RuleSet.copyleftInDependencyRule() = packageRule("COPYLEFT_IN_DEPENDENCY") 
fun RuleSet.copyleftLimitedInDependencyRule() = dependencyRule("COPYLEFT_LIMITED_IN_DEPENDENCY") 
```

Comment (fviernau) [reply evaluator-rules/src/main/resources/evaluator.rules.kts:1022]: I changed it and ditched my preference.

**Review by tsteenbe — APPROVED** [review]

**Pull request merged**